runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -12611,3 +12559,516 @@ J9::X86::TreeEvaluator::andORStringEvaluator(TR::Node *node, TR::CodeGenerator *
+   deps->addPreCondition(j9classReg, TR::RealRegister::NoReg, cg);
+   deps->addPostCondition(j9classReg, TR::RealRegister::NoReg, cg);
+   generateLabelInstruction(LABEL, node, endLabel, deps, cg);
+
+      {
+      TR_OutlinedInstructionsGenerator og(fieldReportLabel, node ,cg);
+      generateReportFieldAccessOutlinedInstructions(node, endLabel, dataSnippet, isWrite, cg);
+      }
+   cg->stopUsingRegister(j9classReg);
+   }
+
+
+void J9::X86::TreeEvaluator::VMrdbarEvaluatorForFieldWatch(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   bool isStatic = node->getSymbolReference()->getSymbol()->getKind() == TR::Symbol::IsStatic;
+   J9Method *owningMethod = (J9Method *)getMethodFromBCInfo(node->getByteCodeInfo(), cg->comp());
+   int32_t bcIndex = node->getByteCodeInfo().getByteCodeIndex();
+   for (int i=0; i < node->getNumChildren(); i++) //evaluate children here to avoid first evaluation point in ICF
+      cg->evaluate(node->getChild(i));
+   if (isStatic)
+      directReadWriteBarrierHelperForFieldWatch(node, owningMethod, bcIndex, false /* isWrite */ , cg);
+   else indirectReadWriteBarrierHelperForFieldWatch(node, owningMethod, bcIndex, false /* isWrite */, cg);
+   }
+
+void J9::X86::TreeEvaluator::VMwrtbarEvaluatorForFieldWatch(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   bool isStatic = node->getSymbolReference()->getSymbol()->getKind() == TR::Symbol::IsStatic;
+   J9Method *owningMethod = (J9Method *)getMethodFromBCInfo(node->getByteCodeInfo(), cg->comp());
+   int32_t bcIndex = node->getByteCodeInfo().getByteCodeIndex();
+   for (int i=0; i < node->getNumChildren(); i++) //evaluate children here to avoid first evaluation point in ICF
+      cg->evaluate(node->getChild(i));
+   if (isStatic)
+      directReadWriteBarrierHelperForFieldWatch(node, owningMethod, bcIndex, true /* isWrite */ , cg);
+   else indirectReadWriteBarrierHelperForFieldWatch(node, owningMethod, bcIndex, true /* isWrite */, cg);
+   }
+
+void J9::X86::TreeEvaluator::rdbarSideEffectsEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   // The child of rdbar for static field is for side affect
+   // and is not used by store evaluator
+   TR::Node *sideEffectNode = node->getSymbolReference()->getSymbol()->isStatic() ? node->getFirstChild(): NULL;
+   if (cg->comp()->getOption(TR_EnableFieldWatch))
+      {
+      TR::TreeEvaluator::VMrdbarEvaluatorForFieldWatch(node, cg);
+      if (sideEffectNode)
+         cg->decReferenceCount(sideEffectNode);
+      }
+   else if (sideEffectNode)
+      cg->recursivelyDecReferenceCount(sideEffectNode);
+   }
+
+TR::Register *J9::X86::TreeEvaluator::irdbarEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR::TreeEvaluator::rdbarSideEffectsEvaluator(node, cg);
+   if (!TR::Compiler->om.shouldGenerateReadBarriersForFieldLoads())
+      return TR::TreeEvaluator::iloadEvaluator(node, cg);
+
+   //sequence for pauseless GC
```

Comment:
The official term for pause-less GC is Concurrent Scavenge.

Reply:
changed the comment to the official term. Thanks!